### PR TITLE
[FLINK-6893][table]Add BIN supported in SQL

### PR DESCRIPTION
**Pull request by sunjincheng121**
In this PR. have Add BIN supported in SQL,e.g.:
```
BIN(12) -> "1100"
BIN(10)-> "1010"
```
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-6893][table]Add BIN supported in SQL")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by sunjincheng121** [comment]
I'll Rebase code and add doc after FLINK-6960 & FLINK-6925 merged.

**Comment by sunjincheng121** [comment]
I had rebase code and add the document.

**Comment by sunjincheng121** [comment]
I have rebase the code. I appreciated if you can review this PR. @wuchong @shaoxuan-wang 

Best, Jincheng

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala [hunk 1]
```diff
@@ -33,6 +33,14 @@ object ScalarSqlFunctions {
     OperandTypes.NILADIC,
     SqlFunctionCategory.NUMERIC)
 
+  val BIN = new SqlFunction(
+    "BIN",
+    SqlKind.OTHER_FUNCTION,
+    ReturnTypes.explicit(SqlTypeName.VARCHAR),
+    null,
+    OperandTypes.NUMERIC,
```
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala]: I don't think BIN accepts all NUMERIC operands. I think it only accepts BIGINT, TINYINT, SMALLINT,INTEGER.  And not support other decimal numeric.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala [hunk 2]
```diff
@@ -352,6 +352,14 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
...
   }
 
+  @Test
+  def testBin(): Unit = {
+    testSqlApi("BIN(12)", "1100")
+    testSqlApi("BIN(10)", "1010")
+    testSqlApi("BIN(0)", "0")
+    testSqlApi("BIN(f32)","1111111111111111111111111111111111111111111111111111111111111111")
+  }
+
```
Comment (wuchong) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala]: Would be better to add validation tests to `ScalarFunctionsValidationTest` to check the expected exception for unsupported operand types.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala (hunk 1 above)
Comment (sunjincheng121) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala]: Yes, In current code we only support BYTE,SHORT,INTEGER,LONG. If we try to using a FLOAT input we will got the EXCEPTION. 
If understand you correctly you want define a new operand types checker, just like `tableAPI(ScalarSqlFunction)`, But I in SQL I'd better using CALCITE's default `OperandTypeChecker`. What do you think?

Comment (haohui) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala]: You can construct a checker easily using `OperandTypes.or()` in Calcite.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/FunctionGenerator.scala [hunk 3]
```diff
@@ -414,6 +415,11 @@ object FunctionGenerator {
     DOUBLE_TYPE_INFO,
     BuiltInMethods.LOG_WITH_BASE)
 
+  addSqlFunction(
```
Comment (haohui) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/FunctionGenerator.scala]: Unnecessary changes?

File: docs/dev/table/sql.md [hunk 4]
```diff
@@ -1691,7 +1691,7 @@ INITCAP(string)
 
     <tr>
       <td>
-        {% highlight text %}
```
Comment (haohui) [inline docs/dev/table/sql.md]: Unnecessary changes.

**Comment by sunjincheng121** [comment]
@thedrow Thanks a lot ...

Best, Jincheng
